documents/description.md

```diff
@@ -2,8 +2,62 @@
+- You need access to a large quantity of sheet music (the second time you go through a piece, you're no longer sight-reading).  Furthermore, this sheet music might need to be highly specific.  A beginner trumpeter, for example, would want single-stave, treble clef sheet music with notes whose pitches range from F3 to G5 and whose rhythms aren't too complex.  A progressive-metal/jazz fusion guitarist might want to specifically train fast pieces with odd time signatures that modulate key frequently.  Finding enough sheet music to suit your specific needs can be difficult and expensive.
+- Musical pieces of specific genres rely on a limited set of patterns and cliches.  Thus, instrumentalists will often find themselves playing the same phrases over and over when sight-reading "new" sheet music. This is good because it allows instrumentalists to quickly recognize common patterns, but it's bad because it's not entirely sight-reading. Practicing this way creates a sight-reading "crutch" that can't always be depended upon.  Who knows when a musician will have to sight-read in a drastically different genre than what they are used to?
+
+The goal of this DSL is to alleviate these issues by allowing users to generate an infinite amount of random sheet music based on highly-customizable parameters.
+
 ## Language domain
 
+The domain of this DSL is parameterized random (is there a word for this?) sheet music generation.  Instrumentalists or vocalists who wish to practice sight-reading/singing would benefit from this DSL.
+There exist a few programs/DSLs in this domain.  
+- [randomsheetmusic.com](http://www.randomsheetmusic.com/) generates random simple sheet music.  It is very easily accessible, but also quite limited.  It only generates single-clef, monophonic melodies and has only a few parameters that users can change.  The simple design (visually) is something I'm after, but I'd of course like to have more expanded functionality in my DSL.
+- [Melisma Stochastic Melody Generator](http://www.link.cs.cmu.edu/melody-generator/) gives a lot more control to users, but is also monophonic and doesn't generate viewable sheet music (only MIDI).  Some of the parameter names are also a bit confusing (rhythmic anchoring, key profile type).  I'd take some cues from some of the parameters, but I'd strive to make them easier to understand for beginner instrumentalists and for young kids (11 to 12-year-olds, maybe).
+- [Impro-visor](https://www.cs.hmc.edu/~keller/jazz/improvisor/) is a flexible program that generates sheet music and MIDI of single-line  improvisations over given chord changes. This program has a level of refinement and sophistication that I hope to achieve, but its focus is slightly different than mine.  It's meant to help with improvising rather than sight-reading, so its more geared toward generating licks than generating customizable sheet music.
```

Comment:
Hmm, I worked on this project. This description isn't very accurate. I think if you want to develop a decent algorithm, it would help to look at the source code for this project.